Emulator/SPD.File.Emulator/Spd/SpdTextureEntry.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections;
+using System.Diagnostics.CodeAnalysis;
+
+namespace SPD.File.Emulator.Spd;
+
+public struct SpdTextureEntry
+{
+#pragma warning disable IDE0044 // Add readonly modifier
+#pragma warning disable IDE0051 // Remove unused private members
+    int _textureId;
+    int _unk04;
+    int _textureDataOffset;
+    int _textureDataSize;
+    int _textureWidth;
+    int _textureHeight;
+    int _unk18;
+    int _unk1c;
+    unsafe fixed byte _textureName[16];
+
+    public readonly int GetTextureId() => _textureId;
+    public readonly (int, int) GetTextureOffsetAndSize() => (_textureDataOffset, _textureDataSize);
+    public void SetTextureOffset(int newOffset) => _textureDataOffset = newOffset;
+    public void SetTextureId(int id) => _textureId = id;
+#pragma warning restore IDE0044 // Add readonly modifier
+#pragma warning restore IDE0051 // Remove unused private members
+}
+
+public class SpdTextureDictionary : IDictionary<int, SpdTextureEntry>
+{
+    private readonly Dictionary<int, SpdTextureEntry> textureDictionary;
+
+
+    public SpdTextureDictionary()
+    {
+        textureDictionary = new Dictionary<int, SpdTextureEntry>();
+    }
+
+    public SpdTextureEntry this[int key]
```

Comment:
This can be simplified to 
```
public SpdTextureEntry this[int key]
{
    get => _textureDictionary[key];
    set => _textureDictionary[key] = value;
}
```
And variable renamed to use underscore per C# style. 😉